mesheryctl/pkg/utils/loader.go

```diff
@@ -0,0 +1,66 @@
+package utils
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+var symbols = []string{"|", "/", "-", "\\"}
+
+type Loader struct {
+	Message string
+	runChan chan struct{}
+	stop    sync.Once
+}
+
+func NewLoader(message string) *Loader {
+	l := &Loader{
+		Message: message,
+		runChan: make(chan struct{}),
+	}
+	return l.Start()
+}
+
+func (l *Loader) Start() *Loader {
+	go l.writer()
+	return l
+}
+
+func (l *Loader) Stop() {
+	l.stop.Do(func() {
+		close(l.runChan)
+		l.clear()
+	})
+	fmt.Printf("\033[2K")
+}
+
+func (l *Loader) animate() {
+	var m string
+	for i := 0; i < len(symbols); i++ {
+		m = l.Message + " " + symbols[i]
+		Log.Info(m)
+		time.Sleep(time.Millisecond * 200)
```

Comment:
😬 `time.Sleep()`

Can this loader be event-driven?

Reply:
Yup, need to create another channel for that